appmap/django.py

```diff
@@ -0,0 +1,46 @@
+from appmap._implementation.event import SqlEvent, ReturnEvent
+from appmap._implementation.recording import Recorder
+
+from django.db.backends.signals import connection_created
+from django.dispatch import receiver
+
+import time
+
+
+class ExecuteWrapper:
+    def __init__(self):
+        self.recorder = Recorder()
+
+    def __call__(self, execute, sql, params, many, context):
+        start = time.monotonic()
+        try:
+            return execute(sql, params, many, context)
+        finally:
+            if self.recorder.enabled:
+                stop = time.monotonic()
+                duration = stop - start
+                # Note this logic is based on django.db.backends.utils.CursorDebugWrapper.
+                if many:
+                    # Sometimes the same query is executed with different parameter sets.
+                    # Instead of substituting them all, just say how many times it was run.
```

Comment:
Can you explain this a little more? I'm not sure there's a change needed for the comment, I'd just like to know what's going on.

Reply:
Sometimes the same SQL will get executed with different parameters, so the arguments will be like eg. `sql="INSERT INTO foo VALUES (%s, %s)", params=[["first", "record"], ["second", "record"]]`. I'd assume that the length of `params` could be significant in this case; `params` could be an iterable without known length, hence the `TypeError` exception.

Reply:
Ah, ok, I understand.

I wonder if there's value in rendering the first few`params`, truncating the string if it's too long?

Reply:
I haven't actually seen this used yet; admittedly, I haven't tried a lot of code, but it could be only rarely used. I just went with what CursorDebugWrapper does; it's what django itself uses to log queries.